client/network/src/discovery.rs

```diff
@@ -536,6 +546,12 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 		addr: &Multiaddr,
 		error: &dyn std::error::Error
 	) {
+		if let Some(peer_id) = peer_id {
+			if let Some(list) = self.ephemeral_addresses.get_mut(peer_id) {
+				list.retain(|a| a != addr);
```

Comment:
Looks like this chunk can be simplified using combinators.

Reply:
It's pretty readable as is.